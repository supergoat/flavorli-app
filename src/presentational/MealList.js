// @flow
import React, { Component } from "react";
import {
  FlatList,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import MealItem from './MealItem';
import FAColors from "../common/FAColors";
import FAStyleSheet from "../common/FAStyleSheet";
import type { MealType } from "../common/FATypes";

/** ============================================================================
<MealList
  mealList={MealType[]}
  onPressMeal: (MealType) => void
/>
--------------------------------------------------------------------------------
Displays a list of meals
============================================================================= */
type Props = {
  mealList: MealType[],
  onPressMeal: (MealType) => void
};
class MealList extends Component<Props> {

  static defaultProps = { mealList: [] };

  render() {
    const { mealList, onPressMeal } = this.props;

    return (
      <FlatList
        style={styles.container}
        data={mealList}
        keyExtractor={item => item.id}
        renderItem={({ item }) => (
          <MealItem
            meal={item}
            onPressMeal={onPressMeal}
          />
        )}
        ListEmptyComponent={() => (
          <Text>No meals available right now. Please try again later</Text>
        )}
      />
    );
  }
}

/* StyleSheet
============================================================================= */
const styles = FAStyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    backgroundColor: FAColors.white
  }
});

/* Export
============================================================================= */
export default MealList;
