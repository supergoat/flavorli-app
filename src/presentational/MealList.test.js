// @flow
import React from 'react';
import { View, Text } from 'react-native';
import MealList from './MealList';
import renderer from 'react-test-renderer';

const mealList = [
  {
    id: '1',
    name: 'Lasagna',
    description: 'This is the lasagna description',
    price: 12.00
  },
  {
    id: '2',
    name: 'Spaghetti Bolognese',
    description: 'This is the Spaghetti Bolognese description',
    price: 10.00
  },
  {
    id: '3',
    name: 'Tiramisu',
    description: 'This is the Tiramisu description',
    price: 5.00
  }
]

describe('MealList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MealList onPressMeal={jest.fn() }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when is a list of meals is passed as a prop', () => {
    const tree = renderer.create(
      <MealList
        mealList={mealList}
        onPressMeal={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
