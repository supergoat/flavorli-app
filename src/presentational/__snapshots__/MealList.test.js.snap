// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealList renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 10,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <Text>
      No meals available right now. Please try again later
    </Text>
  </View>
</RCTScrollView>
`;

exports[`MealList renders correctly when is a list of meals is passed as a prop 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  data={
    Array [
      Object {
        "description": "This is the lasagna description",
        "id": "1",
        "name": "Lasagna",
        "price": 12,
      },
      Object {
        "description": "This is the Spaghetti Bolognese description",
        "id": "2",
        "name": "Spaghetti Bolognese",
        "price": 10,
      },
      Object {
        "description": "This is the Tiramisu description",
        "id": "3",
        "name": "Tiramisu",
        "price": 5,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 10,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "opacity": 1,
            "padding": 10,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333C4E",
              "fontFamily": "Lato",
              "fontSize": 18,
            }
          }
        >
          Lasagna
        </Text>
        <Text
          style={
            Object {
              "color": "#7e8890",
              "fontFamily": "Lato",
              "fontSize": 16,
              "paddingVertical": 10,
            }
          }
        >
          This is the lasagna description
        </Text>
        <Text
          style={
            Object {
              "color": "#788087",
              "fontFamily": "Lato",
              "fontSize": 16,
            }
          }
        >
          £
          12.00
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "opacity": 1,
            "padding": 10,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333C4E",
              "fontFamily": "Lato",
              "fontSize": 18,
            }
          }
        >
          Spaghetti Bolognese
        </Text>
        <Text
          style={
            Object {
              "color": "#7e8890",
              "fontFamily": "Lato",
              "fontSize": 16,
              "paddingVertical": 10,
            }
          }
        >
          This is the Spaghetti Bolognese description
        </Text>
        <Text
          style={
            Object {
              "color": "#788087",
              "fontFamily": "Lato",
              "fontSize": 16,
            }
          }
        >
          £
          10.00
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "opacity": 1,
            "padding": 10,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333C4E",
              "fontFamily": "Lato",
              "fontSize": 18,
            }
          }
        >
          Tiramisu
        </Text>
        <Text
          style={
            Object {
              "color": "#7e8890",
              "fontFamily": "Lato",
              "fontSize": 16,
              "paddingVertical": 10,
            }
          }
        >
          This is the Tiramisu description
        </Text>
        <Text
          style={
            Object {
              "color": "#788087",
              "fontFamily": "Lato",
              "fontSize": 16,
            }
          }
        >
          £
          5.00
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
